src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/Publisher.java

```diff
@@ -33,13 +37,16 @@
   private final long initTime;
 
   private Collator collator;
+  private FlipFlopDetector flipFlopDetector;
   private boolean isMuted = false;
   private Map<String, Long> actionToExecutionTime;
 
   public Publisher(int evalIntervalSeconds, Collator collator) {
     super(0, evalIntervalSeconds);
     this.collator = collator;
     this.actionToExecutionTime = new HashMap<>();
+    // TODO please bring in guice so we can configure this with DI
```

Comment:
Might want to update this TODO.

Reply:
DI refers to dependency injection in this context